nautobot/utilities/fields.py

```diff
@@ -72,3 +81,108 @@ def deconstruct(self):
             [self.target_field],
             kwargs,
         )
+
+
+class JSONArrayField(models.JSONField):
+    """
+    An ArrayField implementation backed JSON storage.
+    Replicates ArrayField's base field validation.
+    """
+
+    _default_hint = ("list", "[]")
+
+    def __init__(self, base_field, **kwargs):
+        self.base_field = base_field
+        super().__init__(**kwargs)
+
+    def set_attributes_from_name(self, name):
+        super().set_attributes_from_name(name)
+        self.base_field.set_attributes_from_name(name)
+
+    @property
+    def description(self):
+        return "JSON Array of %s" % self.base_field.description
+
+    def get_prep_value(self, value):
+        """Perform preliminary non-db specific value checks and conversions."""
+        if value is not None:
+            if not isinstance(value, (list, tuple)):
+                raise ValueError("value {} is not list or tuple".format(value))
+            value = [self.base_field.get_prep_value(v) for v in value]
+        return super().get_prep_value(value)
+
+    def deconstruct(self):
+        """
+        Return enough information to recreate the field as a 4-tuple:
+         * The name of the field on the model, if contribute_to_class() has
+           been run.
+         * The import path of the field, including the class:e.g.
+           django.db.models.IntegerField This should be the most portable
+           version, so less specific may be better.
+         * A list of positional arguments.
+         * A dict of keyword arguments.
+        """
+        name, path, args, kwargs = super().deconstruct()
+        kwargs.update(
+            {
+                "base_field": self.base_field.clone(),
+            }
+        )
+        return name, path, args, kwargs
+
+    def to_python(self, value):
+        """
+        Convert |value| into JSON, raising django.core.exceptions.ValidationError
+        if the data can't be converted. Return the converted value.
+        """
+        if isinstance(value, str):
+            # Assume we're deserializing
+            vals = json.loads(value)
```

Comment:
Do we need to catch a potential `JSONDecodeError` here and map it to a `ValidationError`?